### ISSUE TITLE
CachingModelPropertiesProvider - NullPointerException after lazy load configuration

### ISSUE THREAD

**Issue by lucasoares** [springfox/springfox#2992]
I'm facing the issue #1207 too, now with newer versions of springfox.

Springfox Versions:
![image](https://user-images.githubusercontent.com/10624972/57642469-c8ba4580-758d-11e9-92fa-5f19e045913a.png)

Spring Version: `2.1.2-RELEASE` (using spring cloud parent `Greenwich.RELEASE`.

I use `undertow`  and everything works without setting all beans to lazy init.

After setting all spring beans to lazy load to improve the startup time, swagger is reporting error calculating properties for models. The cause is a NullPointerException because the objectMapper are null.

There is any kind of workaround this springfox issue? Or even a better way to set lazy initialization that works with swagger.

I have a `BeanFactoryPostProcessor` implemented that loop for all beans setting it to lazy init.

Thank you.

**Comment by chris-pccw** [comment]
I am having the same issue too with spring-boot 1.5.22.RELEASE, and springfox-swagger2 version 2.9.2

**Comment by previousdeveloper** [comment]
+1

**Comment by ledjon** [comment]
Yeah with spring-boot really advertising lazy initialization as of 2.2 this issue really needs to be resolved.

**Comment by nulligor** [comment]
+1

**Comment by juniorrhis** [comment]
+1

**Comment by manoelstilpen** [comment]
+1

**Comment by EstevaoResende** [comment]
+1

**Comment by cezar-tech** [comment]
Also having this

**Comment by fapostolo** [comment]
All my models appear empty, when I put the bare bones:

new Docket(DocumentationType.SWAGGER_2)
                .select()
                .apis(RequestHandlerSelectors.any())
                .paths(PathSelectors.any())
                .build()

I got some new ones from Spring itself

2020-01-01 07:27:25.368  WARN 5972 --- [  restartedMain] s.d.s.p.CachingModelPropertiesProvider   : Exception calculating properties for model(org.springframework.web.servlet.ModelAndView) -> ModelContext{groupName=default, type=org.springframework.web.servlet.ModelAndView, isReturnType=true}. java.lang.NullPointerException
2020-01-01 07:27:25.372  WARN 5972 --- [  restartedMain] s.d.s.p.CachingModelPropertiesProvider   : Exception calculating properties for model(org.springframework.web.servlet.ModelAndView) -> ModelContext{groupName=default, type=org.springframework.web.servlet.ModelAndView, isReturnType=true}. java.lang.NullPointerException

**Comment by alexZaicev** [comment]
Same hear as well in the logs and swagger v2 documentation models appear to be empty

**Comment by lizhuangs** [comment]
+1

**Comment by sequarius** [comment]
+1,have the same problem.

**Comment by oaichina** [comment]
+1,have the same problem.Hope to fix it as soon as possible.

**Comment by dilipkrish** [comment]
@oaichina  Is this happening in 3.0.0, I suspect the answer is yes, but just in case

**Comment by ilmn-aeolus** [comment]
@dilipkrish - I just tried with spring-fox 3.0.0, and the problem is still there.

**Comment by varnem2** [comment]
I just encountered this issue, but I don't want to just do a plus one so I'll add additional information. It seems this has been a problem for a long time, here's two additional stale issues with what I believe to be the same problem with clear descriptions. (Don't forget google chrome can translate the page for you if you right click. :D)

https://github.com/springfox/springfox/issues/3182
https://github.com/springfox/springfox/issues/2925

**Comment by guisesterheim** [comment]
@dilipkrish 
I was facing the same issue, even after updating to springfox 3. 

The scenario was that I had recently added the configuration "spring.main.lazy-initialization: true" in my application props for testing purposes. 
When I removed it, the exception stopped happening and all bodies are back there. 
@lucasoares can you check if you're using this parameter?

Hope it has helped

**Comment by lucasoares** [comment]
@guisesterheim this exactly what I explained on my description: we can't have lazy initialization if we want to use springfox.

This is exactly what we need to fix here. I want lazy initialization and springfox to work together.

**Comment by wushp** [comment]
+1,have the same problem.

**Comment by newpanjing** [comment]
+1,have the same problem.

**Comment by ivyson10** [comment]
> @dilipkrish
> I was facing the same issue, even after updating to springfox 3.
> 
> The scenario was that I had recently added the configuration "spring.main.lazy-initialization: true" in my application props for testing purposes.
> When I removed it, the exception stopped happening and all bodies are back there.
> @lucasoares can you check if you're using this parameter?
> 
> Hope it has helped

I use it, now it's working, thanks

**Comment by lucasoares** [comment]
> > @dilipkrish
> > I was facing the same issue, even after updating to springfox 3.
> > The scenario was that I had recently added the configuration "spring.main.lazy-initialization: true" in my application props for testing purposes.
> > When I removed it, the exception stopped happening and all bodies are back there.
> > @lucasoares can you check if you're using this parameter?
> > Hope it has helped
> 
> I use it, now it's working, thanks

Hello!

Just a reminder because disabling this property is exactly what I don't want to do. Disabling it is not the solution for this issue.

Springfox should work with lazy initialization!

**Comment by ctobe-zen** [comment]
Any updates on this getting fixed?

**Comment by lucasoares** [comment]
> Any updates on this getting fixed?

To be honest the update is this: https://springdoc.org/#migrating-from-springfox

**Comment by codymikol** [comment]
It seems like this project should be archived, I don't think it is in development anymore for anyone looking for a solution to this problem.

**Comment by extremelogic-ph** [comment]
Just wondering if we just set `@Lazy(false)` somewhere, but where? This approach allows us to retain `lazy-initialization: true`

**Comment by lucasoares** [comment]
This project should be archived. I suggest using the following project: https://github.com/springdoc/springdoc-openapi

A migration guide to leave springfox: https://springdoc.org/#migrating-from-springfox

**Comment by xqdd** [comment]
@extremelogic-ph 
> Just wondering if we just set `@Lazy(false)` somewhere, but where? This approach allows us to retain `lazy-initialization: true`



```application.yaml
spring.main.lazy-initialization: true
```

```java
@Configuration
@Lazy(value = false)
public class EagerLoadConfig {
    @Autowired
    private RequestMappingHandlerAdapter adapter;
}
```
